### PR TITLE
DCJ-676: Remove setupProxy and http-proxy-middleware

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-676

### Summary
This PR removes unnecessary configurations for local development.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by fboulnois — APPROVED** [review]
nice cleanup 👍

File: DEVNOTES.md [hunk 1]
```diff
@@ -12,8 +12,8 @@ nvm install 22.6.0
 npm install
 ```
 
-3. Install configs for an environment. This example is for the `alpha` environment, but you can use values from any environment by looking at the deployed configs in https://duos-k8s.dsde-{%ENV%}.broadinstitute.org/config.json where {%ENV%} is any of `dev`, `staging`, `alpha`, or `prod` 
-Remember to set the `env` value appropriately. We use `local` for running via npm, but under docker, we use a real env like `dev`.
+3. Install configs for an environment. This example is for the `alpha` environment, but you can use values from any environment by looking at the deployed configs in https://duos-k8s.dsde-{%ENV%}.broadinstitute.org/config.json where {%ENV%} is any of `dev`, `staging`, `alpha`, or `prod`
+Remember to set the `env` value appropriately, for example, `dev`. Certain features are not available on different environments and this will simulate that for local development.
```
Comment (rjohanek) [inline DEVNOTES.md]: is the point here that we want to point to dev backend for local development? Does "this" in this sentence refer to dev? This is a little unclear to me

**Review by rjohanek — APPROVED** [review]

File: DEVNOTES.md (hunk 1 above)
Comment (rushtong) [reply DEVNOTES.md]: Thanks for the question Rae! `this` refers to the `env` variable. I'll take another stab at this language.

Comment (rjohanek) [reply DEVNOTES.md]: great thanks looks good!

**Pull request merged**